src/clients/SpokePoolClient.ts

```diff
@@ -334,6 +336,21 @@ export class SpokePoolClient extends BaseAbstractClient {
     return fills?.find((fill) => validateFillForDeposit(fill, deposit));
   }
 
+  private isEvmRepaymentValid(fill: Fill, matchedDeposit: Deposit): boolean {
+    // Slow fills don't result in repayments so they're always valid.
+    if (isSlowFill(fill)) {
+      return true;
+    }
+    // Lite chain deposits force repayment on origin chain.
+    const repaymentChainId = matchedDeposit.fromLiteChain ? fill.originChainId : fill.repaymentChainId;
+    // Return undefined if the requested repayment chain ID is not recognized by the hub pool.
+    const possibleRepaymentChainIds = this.configStoreClient?.getEnabledChains();
+    if (isDefined(possibleRepaymentChainIds) && !possibleRepaymentChainIds.includes(repaymentChainId)) {
+      return false;
+    }
+    return chainIsEvm(repaymentChainId) && !isValidEvmAddress(fill.relayer);
```

Comment:
For forwards compatibility, you should probably do `return !chainIsEvm(repaymentChainId) || (chainIsEvm(repaymentChainId) && isValidEvmAddress(fill.relayer))`. Note that it should be `isValidEvmAddress(fill.relayer)`.

Reply:
https://github.com/across-protocol/sdk/pull/853/commits/6a7d527829360d11554faf52d5f1d1285795c84c

Reply:
I think this is equivalent and also less verbose: 3f821ad3eedef8252dc6b8cb6aa15810a6754afa wdyt